File: minseok/10845/10845/10845.cpp
#include <iostream>
#include <queue>
#include <string>

using namespace std;

int main() {
	int T;
	string command;
	queue<int> queue;

	cin >> T;

	for (int i = 0; i < T; i++) {
		cin >> command;

		if (command == "push") {
			int data;
			cin >> data;
			queue.push(data);
		}

		else if (command == "pop") {
			if (queue.empty()) cout << "-1" << endl;
			else {
				cout << queue.front() << endl;
				queue.pop();
			}
		}

		else if (command == "size")
			cout << queue.size() << endl;

		else if (command == "empty")
			cout << queue.empty() << endl;

		else if (command == "front") {
			if (queue.empty()) cout << "-1" << endl;
			else cout << queue.front() << endl;
		}

		else if (command == "back") {
			if (queue.empty()) cout << "-1" << endl;
			else cout << queue.back() << endl;
		}
	}
}

File: yeonghwan/algorithm-boj/sep/10845.py
from sys import stdin
from collections import deque
input = stdin.readline

N = int(input())

queue = deque()

for _ in range(N):
    command = list(map(str, input().split()))

    if command[0] == "push":
        queue.append(command[1])

    elif command[0] == "pop":
        if len(queue) == 0:
            print(-1)
        else:
            print(queue.popleft())
    elif command[0] == "size":
        print(len(queue))
    elif command[0] == "empty":
        print(1) if len(queue) == 0 else print(0)
    elif command[0] == "front":
        print(queue[0]) if len(queue) != 0 else print(-1)
    elif command[0] == "back":
        print(queue[-1]) if len(queue) != 0 else print(-1)
